extension/src/popup/components/sendPayment/SendAmount/index.tsx

```diff
@@ -48,13 +48,31 @@ import { ScamAssetWarning } from "popup/components/WarningMessages";
 import { TX_SEND_MAX } from "popup/constants/transaction";
 
 import "../styles.scss";
+import { TokenBalances } from "@shared/api/types";
 
 enum AMOUNT_ERROR {
   TOO_HIGH = "amount too high",
   DEC_MAX = "too many decimal digits",
   SEND_MAX = "amount higher than send max",
 }
 
+const getAssetDecimals = (
+  asset: string,
+  balances: TokenBalances,
+  isToken: boolean,
+) => {
+  if (isToken) {
+    const contractId = asset.split(":")[1];
+    const balance = balances.find(({ contractId: id }) => id === contractId);
+
+    if (balance) {
+      return Number(balance.decimals);
+    }
+  }
+
+  return 7;
```

Comment:
I think it'd be nice to save `7` as a const with a little comment explaining where it comes from

Reply:
yeah good call, always hard to glean what these magic numbers are.

Reply:
https://developers.stellar.org/docs/fundamentals-and-concepts/stellar-data-structures/assets#amount-precision